Buddies/Activities/ActivityDescriptionController.swift

```diff
@@ -0,0 +1,108 @@
+    @IBOutlet var contentView: UIView!
+
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var dateLabel: UILabel!
+    @IBOutlet weak var locationLabel: UILabel!
+    @IBOutlet weak var descriptionLabel: UILabel!
+
+    @IBOutlet weak var joinButton: UIButton!
+
+    @IBOutlet weak var topicsArea: UICollectionView!
+
+    @IBOutlet weak var usersArea: UICollectionView!
+
+    var topics: [Topic] = []
+    var users: [User] = []
+
+    var memberStatus: MemberStatus = .none
+    
+    var curActivity: Activity?
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+    }
+
+    // Refreshes the UI elements with new data:
+    func render(withActivity activity: Activity, withUsers users: [User], withMemberStatus status: MemberStatus, withTopics topics: [Topic] ) {
+        self.curActivity = activity
+        registerCollectionViews()
+        joinButton?.layer.cornerRadius = 5
+        self.locationLabel.text = "Location"
+        self.titleLabel.text = activity.title
+        self.descriptionLabel.text = activity.description
+        
+        self.dateLabel.text = activity.timeRange.rangePhrase(relativeTo: Date()).capitalized
+        self.topics = topics
+        self.users = users
+        self.memberStatus = status
+        self.topicsArea.reloadData()
+        self.usersArea.reloadData()
+    }
+    
+    // Returns the number of topics or users for their collections
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if (collectionView === self.topicsArea) {
+            return topics.count
+        } else if (collectionView === self.usersArea){
+            return users.count
+        } else {
+            return 0
+        }
+    }
+    
+    // Returns the correct cell for users and topics
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if (collectionView === self.topicsArea) {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "topic_cell", for: indexPath) as! ActivityTopicCollectionCell
+            let topic = topics[indexPath.row]
+            cell.render(withTopic: topic)
+            return cell
+        } else if (collectionView === self.usersArea){
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "user_cell", for: indexPath) as! ActivityUserCollectionCell
+            let user = users[indexPath.row]
+            let isIndividualOwner = self.curActivity?.getMemberStatus(of: user.uid) == .owner
+            let isCurUserOwner = self.memberStatus == .owner
+            cell.render(withUser: users[indexPath.row], isCurUserOwner: isCurUserOwner, isIndividualOwner: isIndividualOwner)
+            return cell
+        } else {
+            return UICollectionViewCell()
```

Comment:
Is this just a safety net/making the compiler happy?

Reply:
Yes - you have to return something from the function. There are specific things that happen for the collections on the page, so this code path never gets hit